projects/packages/videopress/src/client/admin/components/edit-video-details/use-edit-details.ts

```diff
@@ -61,47 +61,79 @@ export default () => {
 	const { data: video, isFetching } = useVideo( Number( videoId ) );
 
 	const [ updating, setUpdating ] = useState( false );
-	const [ data, setData ] = useState( video );
+
+	const [ data, setData ] = useState( {
+		title: video?.title,
+		description: video?.description,
+		caption: video?.caption,
+	} );
 
 	const updateData = newData => {
 		setData( current => ( { ...current, ...newData } ) );
 	};
 
-	const posterEditData = usePosterEdit( { video } );
-	const metaEditData = useMetaEdit( { videoId, video, data, updateData } );
+	const { selectedTime, updatePosterImage, ...posterEditData } = usePosterEdit( { video } );
+	const { metaChanged, handleMetaUpdate, ...metaEditData } = useMetaEdit( {
+		videoId,
+		video,
+		data,
+		updateData,
+	} );
 
-	const saveDisabled = metaEditData.metaChanged === false && posterEditData.selectedTime === null;
+	const saveDisabled = metaChanged === false && selectedTime === null;
 
 	const handleSaveChanges = () => {
 		setUpdating( true );
 
-		const promises = [ posterEditData.updatePosterImage(), metaEditData.handleMetaUpdate() ];
+		const promises = [ updatePosterImage(), handleMetaUpdate() ];
 
 		// TODO: handle errors
 		Promise.allSettled( promises ).then( results => {
 			const [ posterResult ] = results;
-			const posterImage = posterResult?.value ?? data?.posterImage;
-			const videoData = { ...data, posterImage };
+			const posterImage = posterResult?.value ?? video?.posterImage;
+			const videoData = { ...video, ...data, posterImage };
 
 			setUpdating( false );
 			dispatch?.setVideo( videoData );
 			navigate( '/' );
 		} );
 	};
-	// Make sure we have the latest data from the API
-	// Used to update the data when user navigates direct from Media Library
-	// removed -> https://github.com/Automattic/jetpack/issues/26574
-	// @todo: move state login to the redux store
-	// useEffect( () => {
-	// 	setData( video );
-	// }, [ video ] );
+
+	// Update the data when user enter directly to the edit page
+	// This moment we fetch the video data and update after fetching
+
+	const initialLoading =
+		isFetching &&
+		data?.title === undefined &&
+		data?.description === undefined &&
+		data?.caption === undefined;
+
+	useEffect( () => {
+		let mounted = true;
+
+		if ( ! initialLoading && mounted ) {
+			setData( {
+				title: video?.title,
+				description: video?.description,
+				caption: video?.caption,
+			} );
+		}
+
+		// Avoid updating state if component is unmounted
+		// From: https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies
+		return () => {
+			mounted = false;
+		};
+	}, [ initialLoading ] );
```

Comment:
what's the purpose of this?

Reply:
If the user arrives from media library, we fetch the video info, since it's not in the state, in that moment the video arrives empty, after the fetch we need to update the [local data](https://github.com/Automattic/jetpack/pull/26631#discussion_r987887417).

Reply:
It sounds like a hack, a product of the inconsistency of the implementation. The data is already there, either by the initial state (in case the video belongs to the first page) of through the resolver.

I leave a video that shows the data is there, and how to pull it.


https://user-images.githubusercontent.com/77539/194073321-0b201439-0eff-4d45-87d9-0060063fb84c.mov

Reply:
In the video above, the data gets thanks to the resolver. But it gets even more smooth when the video belongs to the first page of the video list:

https://user-images.githubusercontent.com/77539/194074103-d77d55aa-26de-480b-9133-060947767d83.mov

No placeholders in the video above thanks to the initial state, provided by the backend.

Reply:
Just in case, I commented the code when testing:

<img width="746" alt="image" src="https://user-images.githubusercontent.com/77539/194074696-002c402f-daa4-484e-95ef-c4844c906c17.png">

You can see the `commented` text in the logs

<img width="87" alt="image" src="https://user-images.githubusercontent.com/77539/194074786-5b1f4667-2750-4c1e-a69f-299a8a993dcc.png">